server/request_handler.go

```diff
@@ -0,0 +1,173 @@
+package server
+
+import (
+	"crypto/ecdsa"
+	"encoding/json"
+	"fmt"
+	"github.com/flashbots/rpc-endpoint/types"
+	"github.com/flashbots/rpc-endpoint/utils"
+	"github.com/google/uuid"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"sync"
+	"time"
+)
+
+type RequestHandler interface {
+	ProcessRequest() *types.JsonRpcResponse
+}
+
+// RPC request handler for a single/ batch JSON-RPC request
+type RpcRequestHandler struct {
```

Comment:
I wonder if there is even a need for this struct or whether we could do all in a function.